### PR TITLE
Feature/index

### PR DESCRIPTION
**Pull request by pyakushin**

### CODE REVIEW

File: src/storage/builder.rs [hunk 1]
```diff
@@ -92,26 +92,21 @@ impl Builder {
...
-                "blob file format: {}.{{}}.blob",
-                self.config.blob_file_name_prefix.as_ref().unwrap()
-            );
         }
         self
     }
+
+    /// # Description
+    /// Sets key size limit
+    pub fn key_size(mut self, key_size: u64) -> Self {
```
Comment (agend) [inline src/storage/builder.rs]: `u64` is to huge for key size) `u8` is too small and `u16` seams to be fine)

File: src/storage/core.rs [hunk 2]
```diff
@@ -137,11 +138,21 @@ impl Storage {
...
-    pub async fn write(self, record: Record) -> Result<()> {
+    /// ```no-run
+    /// block_on(async {
+    ///     let key = 42u64.to_be_bytes().to_vec();
+    ///     let data = b"async written to blob".to_vec();
+    ///     await!(storage.write(key, data))
+    /// )};
+    /// ```
+    pub async fn write(self, key: Vec<u8>, data: Vec<u8>) -> Result<()> {
+        if key.len() as u64 != self.shared.config.key_size.ok_or(Error::Unitialized)? {
```
Comment (agend) [inline src/storage/core.rs]: Опечатка `Unitialized`?

File: src/storage/core.rs [hunk 3]
```diff
@@ -137,11 +138,21 @@ impl Storage {
...
+            return Err(Error::KeySizeMismatch(key.len()));
```
Comment (agend) [inline src/storage/core.rs]: Checking for key size can be put in function. May be it make sense to create some trait for key and not to pass `Vec` every where and we can implement for some basic types (`u64` for example)

File: src/storage/builder.rs (hunk 1 above)
Comment (pyakushin) [reply src/storage/builder.rs]: Agree

File: src/storage/core.rs (hunk 2 above)
Comment (pyakushin) [reply src/storage/core.rs]: Fixed

**Pull request merged**